### PR TITLE
feat(events): series admin GET + cadence-drift detection

### LINKED ISSUE #376: Recurring events: cadence-change drift detection / rebase-series endpoint
## Context

When an organiser changes the recurrence cadence on a live series via `PATCH /event-series/{id}/recurrence`, the backend moves the generation cursor forward so the next beat-run materializes dates under the new rule. Already-scheduled future occurrences keep their old dates (this is intentional — sold tickets can't be moved).

The result is an admin dashboard that mixes "old cadence" and "new cadence" rows. The organiser has to eyeball which rows don't match the current rule and cancel them individually.

## Frontend plan (interim)

See `revel-frontend/RECURRING_EVENTS_PLAN.md`, open decision #5. Short-term FE shows a generic, dismissible banner after a cadence PATCH: *"You just changed the cadence. Upcoming occurrences scheduled under the previous rule still appear below — cancel them individually if needed."*

We deliberately did **not** implement client-side RFC 5545 matching to detect drift row-by-row:
- Implementing a correct RRULE matcher in JS is a rabbit hole (BYSETPOS, timezone edge cases, eventual DST support).
- Any JS matcher we write would eventually diverge from the Python `dateutil.rrule` semantics used server-side, producing false positives/negatives on the drift banner.

## Ask (backend)

One of the following would unblock a better UX:

1. **Drift detection endpoint** — `GET /organization-admin/{slug}/event-series/{id}/drift` returns `{ stale_occurrences: [event_id, …] }`, computed server-side against the current rule. FE highlights those rows and can surface a "Cancel all stale dates" bulk action.

2. **Rebase-series endpoint** — `POST /organization-admin/{slug}/event-series/{id}/rebase` (with safety flags, e.g. `cancel_sold_ticket_events: bool`) that atomically cancels old-cadence occurrences and regenerates under the new rule. More opinionated, handles the common case in one click.

3. **Bulk-cancel** — `POST /organization-admin/{slug}/event-series/{id}/cancel-occurrences` accepting a list of `occurrence_date[]`. Cheapest; FE still has to compute which dates to send, so only useful in combination with (1).

(1) is the minimum useful thing and pairs naturally with the existing per-occurrence cancel endpoint. (2) is the "do what I mean" flow but has sharp edges around sold tickets.

## Also related

- Reinstate-cancelled-date flow — no backend support today. FE workaround: clicking an `exdates` chip opens the one-off event creation form with the date prefilled and the series pre-attached. Not a true reinstate (the resulting event is `is_modified=true` from birth). A first-class "remove from exdates and regenerate" endpoint would be cleaner.

Tagging @biagiodistefano for context from the FE plan.

### LINKED ISSUE #377: Recurring events: add GET endpoint for admin recurrence detail
## Problem

There is no `GET` endpoint that returns `EventSeriesRecurrenceDetailSchema`. Today that schema is only returned by **mutations** in the recurring-events controller:

- `POST /organization-admin/{slug}/create-recurring-event`
- `PATCH /organization-admin/{slug}/event-series/{series_id}/template`
- `PATCH /organization-admin/{slug}/event-series/{series_id}/recurrence`
- `POST /organization-admin/{slug}/event-series/{series_id}/cancel-occurrence`
- `POST /organization-admin/{slug}/event-series/{series_id}/pause`
- `POST /organization-admin/{slug}/event-series/{series_id}/resume`

Other admin endpoints return the public-facing `EventSeriesRetrieveSchema`:

- `PUT /event-series-admin/{series_id}/` → `EventSeriesRetrieveSchema` (lacks `recurrence_rule`, `template_event`, `exdates`, `last_generated_until`).

## Why this is a blocker

The frontend series admin dashboard (see `revel-frontend/RECURRING_EVENTS_PLAN.md`, Phase 2) needs to render the full admin shape on page load — it reads `recurrence_rule` (for the summary), `template_event` (link target), `exdates` (cancelled-dates chip list), `last_generated_until` (horizon line), `is_active`, `auto_publish`, `generation_window_weeks`.

Without a GET, the FE would have to either:
1. Call an empty `PATCH .../recurrence` with `{}` to pull the shape back (abuses a mutation endpoint; still writes an update row).
2. Render a partial dashboard on load and hydrate on first mutation (bad UX; the organiser sees the full state only after acting).

Neither is acceptable.

## Ask

Add `GET /organization-admin/{slug}/event-series/{series_id}` (or a similarly namespaced path) that returns `EventSeriesRecurrenceDetailSchema`. Gate it on `OrganizationPermission("edit_event_series")` — same guard the mutations use.

Ideally ships alongside or shortly after PR #369 so the FE recurring-events rollout (tracked separately) isn't blocked.

## Interim FE workaround (if needed)

If this GET can't land in the same cycle, the FE will issue `PATCH .../recurrence` with `{}` to fetch the shape. That works because every field on `EventSeriesRecurrenceUpdateSchema` is optional, but it's semantically ugly and we'd prefer not to ship it. Flagging explicitly so reviewers aren't surprised.

### PR DESCRIPTION
**Pull request by biagiodistefano**
## Summary

Two small read-only additions to the recurring-events admin surface, closing two related FE blockers.

- **Closes #377** — `GET /organization-admin/{slug}/event-series/{series_id}` returns `EventSeriesRecurrenceDetailSchema` (full admin shape: rule, template, exdates, `last_generated_until`, window, active/auto-publish flags). The FE dashboard needed this on page load; before, the only way to read the admin shape was to hit `PATCH .../recurrence` with an empty body, which was semantically ugly.
- **Closes #376** — `GET /organization-admin/{slug}/event-series/{series_id}/drift` returns `{ stale_occurrences: [event_id, ...] }` — IDs of future occurrences whose start no longer matches the current RRULE (i.e. cadence-change drift). Computed server-side against `dateutil.rrule` so the FE doesn't have to re-implement RFC 5545 matching.

Both endpoints:
- Gated on `OrganizationPermission("edit_event_series")` (same guard as the existing mutations).
- Use `UserDefaultThrottle` (per-endpoint override of the class-level `WriteThrottle`) since they're reads.
- Return 404 for non-members — no cross-org leakage.

## Drift semantics

A materialized event is classified as drift iff **all** of these hold:

- `is_template=False`
- `start >= now()` (past occurrences can't be rescheduled)
- `status != CANCELLED` (terminal already)
- `is_modified=False` (manually-shifted occurrences were deliberately moved off-cadence; sweeping them into a "cancel all stale dates" bulk action would be wrong)
- `start.astimezone(UTC) not in {rrule_occurrences_in_[earliest, latest]_minus_exdates}`

Comparison is by UTC instant — events and rrule outputs are both normalized before set membership.

## Test plan

- [x] `make check` (ruff format/lint, mypy --strict, migrations, i18n, file-length) — clean
- [x] `make test` — 4221 passed, 15 skipped, 95% coverage
- [x] 19 new tests: 8 controller (detail + drift) + 11 service (drift) covering empty/no-rule, cadence change, is_modified/cancelled/past exclusions, exdate drift, template exclusion, count-exhausted rule, 404s, cross-org isolation

## Summary by CodeRabbit

* **New Features**
  * Organization admins can retrieve detailed recurring event series info.
  * New cadence-drift check returns a list of future occurrences that no longer match the current recurrence pattern.

* **Tests**
  * Added controller tests covering admin series detail and drift endpoints, including access rules.
  * Added unit tests for drift detection covering exclusions, exdates, UTC normalization, and edge cases.

### CODE REVIEW

**Comment by biagiodistefano** [comment]
### Code review

No issues found. Checked for bugs and CLAUDE.md compliance.

🤖 Generated with [Claude Code](https://claude.ai/code)

<sub>- If this code review was useful, please react with 👍. Otherwise, react with 👎.</sub>

File: src/events/tests/test_service/test_recurrence_service_drift.py [hunk 1]
```diff
@@ -0,0 +1,258 @@
...
+        assert event.id in result
+
+    @freeze_time("2026-04-06 00:00:00")
+    def test_rrule_computed_with_utc_normalization(
+        self,
+        active_series: EventSeries,
+    ) -> None:
+        """Events stored in UTC must be comparable against rule instants even
+        when the rule's ``to_rrule()`` yields naive-tz datetimes in the
+        same-instant-different-representation.
```
[the remark this answers is not in the document]
Comment (biagiodistefano) [reply src/events/tests/test_service/test_recurrence_service_drift.py]: Good catch — you're right. Django with `USE_TZ=True` gives `to_rrule()` an aware `dtstart`, so the returned datetimes are aware, just in a different tzinfo representation (e.g. `Europe/Vienna`) than the UTC-stored event `start` fields. That's the actual mismatch this test pins. Rewritten in 5df21a0.
